asmcomp/cmmgen.ml

```diff
@@ -1640,33 +1640,76 @@ let transl_isout h arg dbg = tag_int (Cop(Ccmpa Clt, [h ; arg], dbg)) dbg
 
 (* Build an actual switch (ie jump table) *)
 
-let make_switch arg cases actions dbg =
-  let is_const = function
+let make_switch =
+  let extract_uconstant =
+    function
     (* Constant integers loaded from a table should end in 1,
        so that Cload never produces untagged integers *)
-    | Cconst_int (n, _)
-    | Cconst_pointer (n, _) -> (n land 1) = 1
-    | Cconst_natint (n, _)
-    | Cconst_natpointer (n, _) -> (Nativeint.(to_int (logand n one) = 1))
-    | Cconst_symbol _ -> true
-    | _ -> false in
-  if Array.for_all (fun (expr, _dbg) -> is_const expr) actions then
-    let to_data_item (expr, _dbg) =
-      match expr with
-      | Cconst_int (n, _)
-      | Cconst_pointer (n, _) -> Cint (Nativeint.of_int n)
-      | Cconst_natint (n, _)
-      | Cconst_natpointer (n, _) -> Cint n
-      | Cconst_symbol (s, _) -> Csymbol_address s
-      | _ -> assert false in
-    let const_actions = Array.map to_data_item actions in
+    | Cconst_int     (n, _), _dbg
+    | Cconst_pointer (n, _), _dbg when (n land 1) = 1 ->
```

Comment:
Please use the indentation style enforced by ocp-ident and the root `.ocp-indent` file.